cloudify_ansible/utils.py

```diff
@@ -624,32 +742,139 @@ def get_executable_path(executable, venv):
+                _ctx.node.properties.get('ansible_external_executable_path')
+            )
+        )
+        return False
+    return True
+
+
+def install_galaxy_collections(_ctx,
+                               collections_to_install=None):
     """
         Handle creation of virtual environments for running playbooks.
         The virtual environments will be created at the deployment directory.
        :param _ctx: cloudify context.
-       :param packages_to_install: list of python packages to install
-        inside venv.
        :param collections_to_install: list of galaxy collections to install
         inside venv.
        """
+
+    if collections_to_install:
+        if is_connected_to_internet():
+            venv_path = get_instance(ctx).runtime_properties.get(PLAYBOOK_VENV)
+            collections_location = _get_collections_location(_ctx)
+            _ctx.logger.info("Installing collections {} to path {}".format(
+                str(collections_to_install),
+                collections_location))
+            install_collections_to_venv(venv_path,
+                                        collections_to_install,
+                                        collections_location)
+        else:
+            raise NonRecoverableError('No internet connection.'
+                                      'Do not use galaxy_collections when'
+                                      ' working on the plugin virtualenv.')
+
+
+def install_roles(_ctx,
+                  roles_to_install=None):
+    """
+        Handle creation of virtual environments for running playbooks.
+        The virtual environments will be created at the deployment directory.
+       :param _ctx: cloudify context.
+       :param roles_to_install: list of roles to install
+       """
+
+    if roles_to_install:
+        if is_connected_to_internet():
+            venv_path =\
+                get_instance(ctx).runtime_properties.get(PLAYBOOK_VENV)
+            roles_location = _get_roles_location(_ctx)
+            _ctx.logger.info("Installing roles {} to path {}".format(
+                str(roles_to_install),
+                roles_location))
+            install_roles_to_venv(venv_path,
+                                  roles_to_install,
+                                  roles_location)
+        else:
+            raise NonRecoverableError('No internet connection.'
+                                      'Do not use roles when'
+                                      ' working on the plugin virtualenv.')
+
+
+def install_extra_packages(_ctx,
+                           packages_to_install=None):
+    """
+        Handle creation of virtual environments for running playbooks.
+        The virtual environments will be created at the deployment directory.
+       :param _ctx: cloudify context.
+       :param packages_to_install: list of python packages to install
+        inside venv.
+       """
+
+    if packages_to_install:
+        if is_connected_to_internet():
```

Comment:
`install_packages_to_venv` also checks `is_connected_to_internet` is it also necessary here?